src/GitHub/App.hs

```diff
@@ -0,0 +1,152 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module GitHub.App
+  (
+   , InstallationAuth 
+   , authenticateInstallation
+   , createInstToken
+  ) where
+
+import Universum hiding          (exp, Option)
+import Crypto.Types.PubKey.RSA   (PrivateKey(..))
+import Data.Aeson                (FromJSON (..), withObject, (.:))
+import Data.Default.Class        (def)
+import GitHub.Auth               (Auth (..))
+import Lens.Micro.Platform       (makeLenses, (^.))
+import Network.HTTP.Req          (POST (..), NoReqBody (..), Option, Url, header, https,
+                                   jsonResponse, req, responseBody, runReq, (/:))
+import Data.Time.Clock           (UTCTime (..))
+import Data.Time.Clock.POSIX     (POSIXTime,getPOSIXTime, utcTimeToPOSIXSeconds)
+import Web.JWT                   (Signer(..), JSON, JWTClaimsSet(..), encodeSigned, numericDate, stringOrURI)
+import qualified Prelude as Show (Show, show)
+import qualified  Data.Text as T (takeWhileEnd, takeWhile)
+
+-- | Json Web Token expiration time. Maximun accepted by github is 10 minutes. 
+jwtExpTime :: POSIXTime
+jwtExpTime = 600.0 
+
+-- | Installation key expiration time. It is fixed by github and is equal to 1 hour. 
+instKeyExpTime :: POSIXTime
+instKeyExpTime = 3600.0
+
+-- | Time preserved to be on the safe side. 
+-- if expiration time of installation auth token <= current time + bufferTime 
+-- then installation Auth token shoul be updated. 
+bufferTime :: POSIXTime 
+bufferTime = instKeyExpTime * 0.25
+
+-- | base URL of api
+baseURL :: Text
+baseURL = "api.github.com"
+
+type BaseURL = Text 
+type InstallationId = Text 
+
+
+-- | Github installation Auth token 
+data GTAToken = GTAToken 
+    {
+        instToken  :: !Text,      -- ^ Installation token 
+        expireAt   :: !POSIXTime  -- ^ expiration time 
```

Comment:
Ok, I think this here is an actual time, but why are you using `POSIXTime` instead of `UTCTime`?